rspace/src/test/scala/coop/rchain/rspace/ReplayRSpaceTests.scala

```diff
@@ -243,18 +243,20 @@ trait ReplayRSpaceTests extends ReplayRSpaceTestsBase[String, Pattern, String, S
                            false,
                            peeks = SortedSet(0)
                          )
-        resultProduce1 <- space.produce(channels(1), datum, false)
-        resultProduce2 <- space.produce(channels(0), datum, false)
+        resultProduce1  <- space.produce(channels(1), datum, false)
+        resultProduce2  <- space.produce(channels(0), datum, false)
+        resultProduce2a <- space.produce(channels(0), datum, false)
```

Comment:
This value is unused, just like the other newly introduced in this test.
We should probably add assertions for the newly-captured values.

In general, I'd say this test is quite big and not easy to understand. Maybe having helper methods that assert the result of the produce/consume while they perform the produce/consume would help. I share this for the long term, all suggestions in this paragraph are optional for this PR :)